include/circt/Dialect/HW/InnerSymbolTable.h

```diff
@@ -101,11 +115,17 @@ class InnerSymTarget {
 };
 
 /// A table of inner symbols and their resolutions.
-class InnerSymbolTable {
+/// enable_shared_from_this is used to enable cross-linking between symbol
+/// tables during construction.
+/// Given this, InnerSymbolTable is only legal to construct as a shared_ptr.
```

Comment:
This complexity is a result of the doubly-linked stuff? If so, I don't think it's worth the added complexity.

Reply:
Plus, shared_ptr isn't effective in the presence of pointer cycles.

Reply:
Parent linkage has now been removed.